router_reset_dns.py

```diff
@@ -295,7 +295,23 @@ def open_dns_page(self) -> bool:
                 except TimeoutException:
                     logger.error(f"Timed out waiting for step {step['location']}, skipping...")
                     return False
-
+                if "value" in step.keys():
+                    try:
+                        value_mode = self.driver.find_element(By.ID, step["location"])
+                        value_mode = Select(value_mode)
+                        value_mode.select_by_value(str(step["value"]))
+                    except TimeoutException:
+                        logger.error(f"Timed out waiting for step {step['location']}, skipping...")
+
+            if step["type"] =="frame":
+                try:
+                    self.driver.switch_to.parent_frame()
+#                    fr = self.driver.findElementById(step["location"])
+                    self.driver.switch_to.frame(step["location"])
+                    logger.debug(f"Switched to frame:"+ str(step["location"]))
```

Comment:
if you use f-string notation please use as follows:
`logger.debug(f"Switched to frame {step['location']}")`